src/ragapp/backend/tests/test_engine.py

```diff
@@ -0,0 +1,128 @@
+from unittest.mock import MagicMock, patch
```

Comment:
This test should keep track of the issues related to changing the agent configuration that affects the construction of the chat engine.